src/AWS.Deploy.ServerMode.Client/ServerModeSession.cs

```diff
@@ -0,0 +1,122 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.\r
+// SPDX-License-Identifier: Apache-2.0
+
+using System;
+using System.Diagnostics;
+using System.Net.Http;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+using AWS.Deploy.ServerMode.Client.Exceptions;
+
+namespace AWS.Deploy.ServerMode.Client
+{
+    /// <summary>
+    /// Helper class that allows launching deployment tool in server mode.
+    /// </summary>
+    public class ServerModeSession
+    {
+        private const int TCP_PORT_ERROR = -100;
+
+        private readonly int _startPort;
+        private readonly int _endPort;
+
+        public ServerModeSession(int startPort = 5000, int endPort = 5100)
```

Comment:
Since new .NET web projects default to 5000 I think we should stay well clear from there. What about 10000 to 10100?

Reply:
that works, somewhat lesser chances of collision.

Reply:
The Toolkit will manage its own installation and location of the cli. Can an optional tool path argument be added so that dotnet calls can be made with the  `--toolpath` arg?

Reply:
Added a todo, we will add ability to configure settings such as port, diagnostics in a separate PR.